### PR TITLE
 Generate compilable accessors for container elements of default package type

### PR DESCRIPTION
**Pull request by bamboo**

### CODE REVIEW

File: subprojects/integ-tests/src/test/kotlin/org/gradle/kotlin/dsl/integration/ProjectSchemaAccessorsIntegrationTest.kt [hunk 1]
```diff
@@ -129,41 +117,83 @@ class ProjectSchemaAccessorsIntegrationTest : AbstractPluginIntegrationTest() {
...
-                    withFile("plugin.gradle.kts", """
-                        tasks.register<CustomTask>("customTask")
-                    """)
-                }
+                    val beans = DefaultPolymorphicDomainObjectContainer(
+                        Any::class.java,
+                        DirectInstantiator.INSTANCE
+                    ) { (it as Bean).name }
+
+                    // TODO: schema doesn't expose the registered element as Bean
```
Comment (eskatos) [inline subprojects/integ-tests/src/test/kotlin/org/gradle/kotlin/dsl/integration/ProjectSchemaAccessorsIntegrationTest.kt]: This looks like a bug to me, polymorphic containers are supposed to expose the implementation type in their schema.

File: subprojects/provider/src/test/kotlin/org/gradle/kotlin/dsl/accessors/DefaultPackageTypesTest.kt [hunk 2]
```diff
@@ -32,4 +34,29 @@ class DefaultPackageTypesTest {
             equalTo(listOf("Extension"))
         )
     }
+
+    @Test
+    fun `#importsRequiredBy takes container elements into account`() {
```
Comment (JLLeitschuh) [inline subprojects/provider/src/test/kotlin/org/gradle/kotlin/dsl/accessors/DefaultPackageTypesTest.kt]: ❓  What's the significance of the `#` symbol in the test name?

Comment (bamboo) [reply subprojects/provider/src/test/kotlin/org/gradle/kotlin/dsl/accessors/DefaultPackageTypesTest.kt]: That's meant to signify the name refers to a function (like in javadoc links).

Comment (JLLeitschuh) [reply subprojects/provider/src/test/kotlin/org/gradle/kotlin/dsl/accessors/DefaultPackageTypesTest.kt]: Ah, thanks.

File: subprojects/integ-tests/src/test/kotlin/org/gradle/kotlin/dsl/integration/ProjectSchemaAccessorsIntegrationTest.kt (hunk 1 above)
Comment (eskatos) [inline subprojects/integ-tests/src/test/kotlin/org/gradle/kotlin/dsl/integration/ProjectSchemaAccessorsIntegrationTest.kt]: :x: This won't affect public api consumers as there's no public way to create polymorphic containers. This test should be removed.

**Pull request merged**